project-plan.md

```diff
@@ -24,50 +24,95 @@ for programming because they facilitate exploratory programming and debugging.
+	* Indication of errors
+	* Integration with Eclipse
+	* Syntax highlighting
+* Should-have
+	* Save and load shell state
+* Could-have
+	* Integration with other IDE's (IntelliJ)
+	* Hover over variables to see value, type and others
+	* GDB style interaction with context / environment
+		* Ability to partially change or update environment
+	* Literate programming
+* Won't-have
 
 If the above turns out to be (too) easy, the REPL can be extended into a
 language playground such as the one offered by the Swift programming language.
 If we decide to extend the project, new agreements will be made between us and
 the client.
 
-## Methodologies and tooling
+# Methodologies and tooling
 
 During this project we will work using the SCRUM methodology with weekly
 sprints. Our backlog, current and completed sprints will be managed using
-Trello. Trello will also be used to communicate documents to the client
-and the TU coach.
-
-We will use pull-based development using GitHub. GitHub Issues will be used to
-track issues (all of this will be linked to Trello in one way or another). To
-facilitate this, we will be given our own repository within the Metaborg
-organization. Each of us will then fork this repository to our private GitHub
-accounts. We will also use one private repository to hold all the documents
+[Trello](https://trello.com/b/u2aKQ12y/bachelor-project-spoofax-repl).
+Trello will also be used to communicate documents to the client and the TU coach.
+Sprints will end on friday, coinciding with most deadlines according to the
+planning for this project as mandated by TU Delft.
+
+Each sprint will start with a "sprint planning event", in which user stories
+from the backlog will be assigned to project members, based on their
+prioritization and their feasibility for this sprint.
+Each sprint will end with a meeting in which we discuss to what degree
+we have achieved our sprint planning and how we can improve on that.
+A working demo should always be available at the time of the "sprint review".
+During sprints, the team will hold a daily scrum every day, where
+they will discuss their progress of the previous day and their planning for
+today, as well as any issues that could be detrimental to the weekly sprint goal.
+
+We will use a pull-based development model using GitHub in conjunction with
+the SCRUM methodology. GitHub Issues will be used to
+track issues (all of this will be linked to Trello in one way or another).
+To facilitate this, we will use repositories from our GitHub organization
+[spoofax-shell](https://github.com/spoofax-shell).
+Each of us will then fork these repositories to our private GitHub
+accounts. We will also use one repository to hold all the documents
 (e.g. this document, the research project, et cetera) supporting our project.
 
-Since Spoofax is an already existing project, we will use whichever tools are
+A pull request will be designated a status tag WIP, RFC or RDY:
+
+* WIP: Work in progress
```

Comment:
Suggestion:

* WIP - Work In Progress: a PR tagged WIP is there for feedback; it will still change so it should not be reviewed in-depth.
* RFC - Ready for Comments: a PR tagged RFC is done as far as the implementation goes: the functionality is there, it is tested, documented and only a (thorough) review is left. If there are too many comments that cannot be resolved within reasonable time, the PR should be demoted to WIP.
* RDY - Ready: a PR tagged RDY has been reviewed by at least one other reviewer, who deems it ready for merge. It is left open a short while to give the remaining team member time to review, after which it is merged.